### ISSUE TITLE
Filling header stack value in the parser 

### ISSUE THREAD

**Issue by antoine-bernabeu** [p4lang/behavioral-model#913]
I am using p4c (version: 1.2.0 (SHA ebd446fa BUILD:debug) ) and simple_switch (version: 1.13.0-498202d9).
I wanted to emit a header stack and to set value of the header  stack in the parser, using metadata as a counter. The p4c compiler found no errors but when I run simple_switch with the json file, I got this error.
![image](https://user-images.githubusercontent.com/55462625/85157393-b49a4a80-b25b-11ea-87da-942e6c2653c0.png)

Here the p4 program and the json file.
[example_headerstack_parser.p4.txt](https://github.com/p4lang/p4c/files/4805704/example_headerstack_parser.p4.txt)
[example_headerstack_parser.json.txt](https://github.com/p4lang/p4c/files/4805705/example_headerstack_parser.json.txt)

**Comment by jafingerhut** [comment]
Your `fill_h3` parser state is assigning values to a header stack element that is currently invalid, I believe.  According to the P4_16 language spec, such assignments are effective 'no ops'.

It is fairly unusual for a P4 parser to fill in headers via any means other than extract calls.  It seems to me that the code you have in your parser state `fill_h3` could functionally be done at the beginning of ingress processing, rather than during parsing?

**Comment by antoine-bernabeu** [comment]
Yes it could be done in the ingress processing, but I wanted to use the parser as a "loop" because if we have hundred of headers in the header stack it can be really annoying to do it in the ingress processing. I tried with setting the header as valid in the parser just before assigning value. The result is the same, same error when running simple_switch.

**Comment by jafingerhut** [comment]
I do not know whether the 'annoying' part you have found is 'repeitive code', but if so, it is not uncommon to work around this by using small programs to generate parts of P4 programs.  This article gives one example to demonstrate: https://github.com/jafingerhut/p4-guide/blob/master/code-generation/README.md
